src/aiida/cmdline/commands/cmd_code.py

```diff
@@ -240,29 +241,35 @@ def show(code):
 @options.OVERWRITE()
 @options.SORT()
 @with_dbenv()
-def export(code, output_file, overwrite, sort):
+def export(code, output_file, overwrite, sort, **kwargs):
     """Export code to a yaml file."""
 
-    import yaml
+    other_args = {'sort': sort, **kwargs}
 
-    code_data = {}
-
-    for key in code.Model.model_fields.keys():
-        value = getattr(code, key).label if key == 'computer' else getattr(code, key)
-
-        # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
-        # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
-        if value is not None:
-            code_data[key] = str(value)
+    if output_file is not None:
+        fileformat = output_file.suffix[1:]
```

Comment:
Should probably warn (or even except) if the user specifies an output file without an extension. And the help string of the command should probably say that the format is deduced from the filename. And I think we currnetly only implement yaml for `AbstractCode`, correct? So it would fail for anything but `.yml` (what about `.yaml`, will that fail?). Maybe we should simply hardcode the format format, regardless of the filename specified.

Reply:
I thought a bit about it and didn't find a way how one could check if the user provided a file extension in a general way, as the filename can include (arbitrarily many) periods, and, e.g., `pathlib.Path().suffix` also returns something if a filename without extension is given (namely the filename). So I agree that it makes sense to hard-code it, both the format check, as well as the default format, which I did now in my latest commit. Indeed, `yaml` wasn't supported, but I added a small `_prepare_yaml` wrapper method, to make that possible, as well. I personally also prefer `.yaml` over `.yml` (and files in the `aiida-code-registry` all end with `.yaml`), so I'm inclined to set that as the default. That's different from the previous default behavior, so not possible if this is something we could just change like that.

Reply:
You said you removed the determination of the format but I still see:
```
fileformat = output_file.suffix[1:]
```
Now it seems that you check before that that the filename ends in either `.yml` or `.yaml` and raise otherwise, but why even do this? I would simply hardcode that you always export in YAML (pick the extension .yml or .yaml, whichever is fine) regardless of the filename, even if it contains no extension whatsoever. The extension doesn't really do anything anyway.

Reply:
@GeigerJ2 what about this comment?

Reply:
Ah, I just overlooked that. Fixed now!